crates/net/rpc-builder/src/lib.rs

```diff
@@ -182,13 +185,44 @@ impl RpcServerBuilder {
     /// This consumes the builder and returns a server.
     ///
     /// Note: The server ist not started and does nothing unless polled, See also
-    pub async fn build(self) -> Result<RpcServer, RpcError> {
-        todo!()
+    pub async fn build(self) -> Result<Vec<RpcServer>, RpcError> {
+        let mut servers = Vec::new();
+        let socket_addr = self.http_ws_addr.unwrap();
+        let ips_endpoint = self.ipc_server_path.unwrap();
+
+        if let Some(builder) = self.http_ws_server_config {
+            let http_ws_server = builder.build(socket_addr).await?;
+            servers.push(RpcServer { server: RpcServerEnum::HttpWs(http_ws_server) })
+        } else {
+            let http_ws_server = ServerBuilder::default().build(socket_addr).await?;
+            servers.push(RpcServer { server: RpcServerEnum::HttpWs(http_ws_server) })
+        }
+
+        if let Some(builder) = self.ipc_server_config {
+            let ipc_server = builder.build(ips_endpoint.path())?;
+            servers.push(RpcServer { server: RpcServerEnum::Ipc(ipc_server) })
+        } else {
+            let ipc_server = IpcServerBuilder::default().build(ips_endpoint.path())?;
+            servers.push(RpcServer { server: RpcServerEnum::Ipc(ipc_server) })
+        }
+
+        Ok(servers)
     }
 }
 
+// can't think of better name rn lol
+/// Represents Server types
+pub enum RpcServerEnum {
+    /// http server
+    HttpWs(Server),
+    /// ipc server
+    Ipc(IpcServer),
+}
+
 /// Container type for the configured RPC server(s): http,ws,ipc
-pub struct RpcServer {}
+pub struct RpcServer {
+    server: RpcServerEnum,
+}
```

Comment:
I decided to remove the ```RpcServerEnum``` and change ```RpcServer``` this to 
```rust
pub struct RpcServer {
    /// http server
    http: Option<Server>,
    /// ws server
    ws: Option<Server>,
    /// ipc server
    ipc: Option<IpcServer>,
}
```
so the ```build``` function on ```RpcServerBuilder``` returns ```Result<RpcServer, RpcError>``` not ```Result<Vec<RpcServer>, RpcError>```

Reply:
Also change
 ```rust     
pub async fn start(self, methods: impl Into<Methods>) -> Result<RpcServerHandle, RpcError>
```
to 
 ```rust     
pub async fn start(self, modules: TransportRpcModules) -> Result<RpcServerHandle, RpcError> 
```